pandas/core/indexes/base.py

```diff
@@ -2874,6 +2874,10 @@ def _convert_scalar_indexer(self, key, kind=None):
 
             elif kind in ['loc'] and is_integer(key):
                 if not self.holds_integer():
+                    try:
```

Comment:
really? what is this actually trying to do. We actually have tests that hit the branches here??